service/controller/resource/nodepool/deployment.go

```diff
@@ -0,0 +1,251 @@
+		"osImageSKU":              "stable",                       // azureMachinePool.Spec.Template.Image.Marketplace.SKU,
+		"osImageVersion":          distroVersion,                  // azureMachinePool.Spec.Template.Image.Marketplace.Version,
+		"replicas":                machinePool.Spec.Replicas,
+		"vnetName":                vnetName,
+		"subnetName":              subnetName,
+		"vmSize":                  azureMachinePool.Spec.Template.VMSize,
+		"zones":                   zones,
+		// This should come from the bootstrap operator.
+		"vmCustomData": workerCloudConfig,
+	}
+
+	armTemplate, err := instance.GetARMTemplate()
+	if err != nil {
+		return azureresource.Deployment{}, microerror.Mask(err)
+	}
+
+	d := azureresource.Deployment{
+		Properties: &azureresource.DeploymentProperties{
+			Mode:       azureresource.Incremental,
+			Parameters: key.ToParameters(templateParams),
+			Template:   armTemplate,
+		},
+	}
+
+	return d, nil
+}
+
+func (r Resource) getSubnetName(azureMachinePool *capzexpv1alpha3.AzureMachinePool, azureCluster *capzv1alpha3.AzureCluster) (string, string, error) {
+	for _, subnet := range azureCluster.Spec.NetworkSpec.Subnets {
+		if azureMachinePool.Name == subnet.Name {
+			return azureCluster.Spec.NetworkSpec.Vnet.Name, subnet.Name, nil
+		}
+	}
+
+	return "", "", microerror.Maskf(notFoundError, "there is no allocated subnet for nodepool %#q in virtual network called %#q", azureMachinePool.Name, azureCluster.Spec.NetworkSpec.Vnet.ID)
+}
+
+func (r *Resource) getFailureDomains(azureCluster *capzv1alpha3.AzureCluster, machinePool *capiexpv1alpha3.MachinePool) ([]string, error) {
+	var validFailureDomain bool
+	allowedFailureDomains := azureCluster.Status.FailureDomains.GetIDs()
+	for _, id := range allowedFailureDomains {
+		if id == machinePool.Spec.Template.Spec.FailureDomain {
+			validFailureDomain = true
+		}
+	}
+
+	if !validFailureDomain && len(allowedFailureDomains) > 0 {
+		return []string{}, microerror.Maskf(notAvailableFailureDomain, "nodepool availability zone %#q is not available in the region", *machinePool.Spec.Template.Spec.FailureDomain)
+	}
+
+	return []string{*machinePool.Spec.Template.Spec.FailureDomain}, nil
+}
+
+func (r *Resource) getWorkerCloudConfig(ctx context.Context, storageAccountsClient *storage.AccountsClient, resourceGroupName, storageAccountName, workerBlobName string, encrypterObject encrypter.Interface) (string, error) {
+	encryptionKey := encrypterObject.GetEncryptionKey()
+	initialVector := encrypterObject.GetInitialVector()
+
+	keys, err := storageAccountsClient.ListKeys(ctx, resourceGroupName, storageAccountName, "")
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	if len(*(keys.Keys)) == 0 {
+		return "", microerror.Maskf(executionFailedError, "storage account key's list is empty")
+	}
+	primaryKey := *(((*keys.Keys)[0]).Value)
+	containerName := key.BlobContainerName()
```

Comment:
Use new `key` function for the blob name, since we need a different blob for every node pool.